### DISCUSSION TITLE
How localtopography.m deals with boundary pixels?

### DISCUSSION THREAD

**Discussion by kaihux** [wschwanghart/topotoolbox#33] [Questions]
Hi Wolfgang,

I read the explanatory text in the localtopography.m, it says localtopography adopts the padding behavior of the function imdilate and imerode in matlab. So i searched in [matlab's documentation](https://www.mathworks.com/help/images/processing-pixels-at-image-borders.html), it says for dilation, pixels beyond the image border are assigned the minimum value afforded by the data type, and for erosion, pixels beyond the image border are assigned the maximum value afforded by the data type. I am still unsure how relief is exactly computed on boundary pixels. Could you please clarify on that? What values are padded for NaNs beyond your AOI boundaries and beyond the DEM boundary?

Let's say I use the clip.m to extract the DEM of the headwater catchments from Basin A, because I want to restrict the relief computation within Basin A and not extending beyond the divide into its neighboring basin. I want to know how the relief (type:range) is computed specifically on the boundaries of these head catchments, beyond which are NaNs.

Thanks!
Kai

**Comment by wschwanghart** [comment]
Hi Kai,
nan pixels or any boundary pixels do not affect the results. They are assumed to be infinitly high and thus they won't be able to affect valid pixels.
Cheers (and sorry for the late reply)
Wolfgang
